pkgs/applications/networking/cluster/crc/default.nix

```diff
@@ -0,0 +1,63 @@
+{ lib
+, buildGoModule
+, fetchFromGitHub
+, git
+, stdenv
+, testers
+, crc
+, runtimeShell
+, coreutils
+}:
+
+buildGoModule rec {
+  pname = "crc";
+  version = "2.4.1";
+  gitCommit = "6b954d4";
+
+  src = fetchFromGitHub {
+    owner = "code-ready";
+    repo = "crc";
+    rev = "v${version}";
+    sha256 = "4ckHvIswVwECAKsa/yN9rJSqZb04ZP1Zomq+1UdT1OY=";
+  };
+
+  vendorSha256 = null;
+
+  nativeBuildInputs = [ git ];
+
+  postPatch = ''
+    substituteInPlace pkg/crc/oc/oc_linux_test.go \
+      --replace "/bin/echo" "${coreutils}/bin/echo"
+
+    substituteInPlace Makefile \
+      --replace "/bin/bash" "${runtimeShell}"
+  '';
+
+  preBuild = ''
+    export HOME=$(mktemp -d)
+    export COMMIT_SHA=${gitCommit}
+  '';
+
+  buildPhase = ''
+    runHook preBuild
+    make
+    runHook postBuild
+  '';
+
```

Comment:
```suggestion
```
Isn't this done automatically? Also this omits some arguments to make which normally get added.

Reply:
I can get all the tags and ldflags without too much trouble, we just need to hardcode the CRC version, the Podman version, the OpenShift version into the manifest or just drop them, which will maybe just affect the `crc version` command.

To fix the tests, I need to look at how to patch the main test because it uses `github.com/cucumber/godog` which expects the build binary to be in the `$GOPATH/go/bin` for e2e test.

Or should I just drop buildGoPackage and use mkDerivation ?

Reply:
> I can get all the tags and ldflags without too much trouble, we just need to hardcode the CRC version, the Podman version, the OpenShift version into the manifest or just drop them, which will maybe just affect the `crc version` command.

Please get version from the version variable.

> To fix the tests, I need to look at how to patch the main test because it uses `github.com/cucumber/godog` which expects the build binary to be in the `$GOPATH/go/bin` for e2e test.

we can symlink it to there



> Or should I just drop buildGoPackage and use mkDerivation ?

no, we should use our build function

Reply:
I've been trying all the solutions I found in the nixpkgs repository for a few days now, but without any success so far, the `TestMain` of the `test/e2e/e2e_test.go` keeps running and obviously fails, I don't know where to go at this point except skipping the check phase.

- [pkgs/tools/text/gucci/default.nix](https://github.com/NixOS/nixpkgs/blob/a14424e2fe541fd08a95b639b772f08c88080416/pkgs/tools/text/gucci/default.nix)
- [pkgs/applications/networking/cluster/clusterctl/default.nix](https://github.com/NixOS/nixpkgs/blob/a14424e2fe541fd08a95b639b772f08c88080416/pkgs/applications/networking/cluster/clusterctl/default.nix)
- [pkgs/tools/misc/git-town/default.nix](https://github.com/NixOS/nixpkgs/blob/a14424e2fe541fd08a95b639b772f08c88080416/pkgs/tools/misc/git-town/default.nix)
- [pkgs/tools/filesystems/go-mtpfs/default.nix](https://github.com/NixOS/nixpkgs/blob/a14424e2fe541fd08a95b639b772f08c88080416/pkgs/tools/filesystems/go-mtpfs/default.nix)
- [pkgs/applications/version-management/git-and-tools/gitbatch/default.nix](https://github.com/NixOS/nixpkgs/blob/a14424e2fe541fd08a95b639b772f08c88080416/pkgs/applications/version-management/git-and-tools/gitbatch/default.nix)

Reply:
I currently don't have time to deeply debug this but if you tried to get it working and currently it does not I would suggest to just disable tests for now with a comment why the tests are not working.

Reply:
Just did that, I'll investigate more in the near future.

Reply:
Just did it with a TODO, I will try to look into it in the near future